Assets/SEE/UI/Window/VariablesWindow/VariablesWindowItem.cs

```diff
@@ -0,0 +1,385 @@
+
+            background.GetComponent<UIGradient>().EffectGradient.SetKeys(
+                new Color[] { BackgroundColor, BackgroundColor.Darker(0.3f) }.ToGradientColorKeys().ToArray(),
+                new GradientAlphaKey[] { new(1, 0), new(1, 1) });
+
+            expandIcon = item.transform.Find("Foreground/Expand Icon");
+
+            if (item.TryGetComponent<PointerHelper>(out pointerHelper))
+            {
+                if (children.Count > 0)
+                {
+                    pointerHelper.ClickEvent.AddListener(ToggleChildren);
+                } else if (VariableReference > 0)
+                {
+                    IsExpanded = false;
+                    pointerHelper.ClickEvent.AddListener(e =>
+                    {
+                        if (e.button == PointerEventData.InputButton.Left)
+                        {
+                            queueRetrieveChildren = true;
+                        }
+                    });
+                } else
+                {
+                    expandIcon.gameObject.SetActive(false);
+                }
+            }
+
+
+            UpdateVisibility();
+            UpdateIndent();
+            UpdateExpand();
+        }
+
+        /// <summary>
+        /// Updates the item.
+        /// 
+        /// <para>
+        ///     Retrieves child items if necessary.
+        /// </para>
+        /// </summary>
+        protected override void Update()
+        {
+            base.Update();
+            if (queueRetrieveChildren)
+            {
+                queueRetrieveChildren = false;
+                RetrieveChildren();
+            }
+        }
+
+        /// <summary>
+        /// Destroys this item and its children.
+        /// </summary>
+        private void OnDestroy()
+        {
+            Destroyer.Destroy(item);
+            foreach (VariablesWindowItem child in children)
+            {
+                Destroyer.Destroy(child);
+            }
+        }
+
+        /// <summary>
+        /// Updates the visibility.
+        /// </summary>
+        private void UpdateVisibility()
+        {
+            item.SetActive(IsVisible);
+            children.ForEach(UpdateChildVisibility);
+        }
+
+
+        /// <summary>
+        /// Updates the indent.
+        /// </summary>
+        private void UpdateIndent()
+        {
+            background.localPosition = background.localPosition.WithXYZ(x: Indent * indentShift);
+            foreground.localPosition = foreground.localPosition.WithXYZ(x: Indent * indentShift);
+            children.ForEach(UpdateChildIndent);
+        }
+
+        /// <summary>
+        /// Updates the children.
+        /// </summary>
+        private void UpdateExpand()
+        {
+            expandIcon.DORotate(new Vector3(0, 0, isExpanded ? -180 : -90), duration: 0.5f);
+            children.ForEach(UpdateChildVisibility);
+        }
+
+        /// <summary>
+        ///     Updates the indentation level of all children.
```

Comment:
While it's kind of fitting that this comment is indented, I think it shouldn't be. (Also applies to next few methods)